tests/test_routers.py

```diff
@@ -164,8 +184,8 @@ def test_register_after_accessing_urls(self):
 
 class TestRootView(URLPatternsTestCase, TestCase):
     urlpatterns = [
-        url(r'^non-namespaced/', include(namespaced_router.urls)),
-        url(r'^namespaced/', include((namespaced_router.urls, 'namespaced'), namespace='namespaced')),
+        re_path(r'^non-namespaced/', include(namespaced_router.urls)),
```

Comment:
why not path() instead?

Reply:
I used `re_path` on purpose, to maintain previous tests logic.

In fact its just a mere substitution from `url` to `re_path`, since `django.conf.urls.url` will be deprecate at some time:

> The **`django.conf.urls.url()`** function from previous versions is now available as **`django.urls.re_path()`**. The old location remains for backwards compatibility, without an imminent deprecation. The old **`django.conf.urls.include()`** function is now importable from **`django.urls`** so you can use **`from django.urls import include, path, re_path`** in your URLconfs.

I agree that they can be replaced with a `path` since none of those unittests uses a regex which is not compatible with a `path`.

I can change them if its ok for you.

Reply:
URL is already deprecated and that's why telling to use path() @tomchristie can shed more light here before you proceed.